packages/siwe/src/provider.tsx

```diff
@@ -64,28 +146,67 @@ export const SiweProvider = ({ children, backendUrl, api }: SiweProviderProps) =
         chainId,
         nonce,
       })
-      const signature = await signer.signMessage(message.prepareMessage())
 
-      await signIn({ signature, message })
+      setLoading(true)
+      const signature = await signer.signMessage(message.prepareMessage()).catch((err) => {
+        setError(err)
+        setLoading(false)
+      })
+      if (!signature) {
+        return
+      }
 
-      void getAuth().then((res) => (res.loggedIn ? setLoggedIn(true) : undefined))
+      try {
+        await signIn({ signature, message })
+      } catch (err) {
+        return setError(err)
+      }
+
+      if (signature === '0x') {
+        localStorage.setItem('siweMessage', JSON.stringify(message))
```

Comment:
I think it is not safe to store siweMessage in localstorage here. Maybe backend should give a temporary cookie and keep the secret for us?